jwst/nsclean/nsclean.py

```diff
@@ -0,0 +1,274 @@
+
+    Parameters
+    ----------
+    input_model : data model object
+        science data
+
+    Mask : 2D boolean array
+        input mask that will be updated
+
+    Returns
+    -------
+    Mask : 2D boolean array
+        output mask with additional flags for slit pixels
+    """
+
+    from jwst.extract_2d.nirspec import offset_wcs
+
+    log.info("Finding slit/slitlet pixels")
+
+    # Get the slit-to-msa frame transform from the WCS object
+    slit2msa = input_model.meta.wcs.get_transform('slit_frame', 'msa_frame')
+
+    # Loop over the slits, marking all the pixels within each bounding
+    # box as False (do not use) in the mask.
+    # Note that for 3D masks (TSO mode), all planes will be set to the same value.
+    for slit in slit2msa.slits:
+        slit_wcs = nirspec.nrs_wcs_set_input(input_model, slit.name)
+        xlo, xhi, ylo, yhi = offset_wcs(slit_wcs)
+        Mask[..., ylo:yhi, xlo:xhi] = False
+
+    return Mask
+
+    
+def create_mask(input_model, n_sigma):
+    """Create the pixel mask needed for setting which pixels to use
+    for measuring 1/f noise.
+
+    Parameters
+    ----------
+    input_model : data model object
+        science data
+
+    n_sigma : float
+        sigma threshold for masking outliers
+
+    Returns
+    -------
+    Mask : 2D boolean array
+        image mask
+
+    nan_pix : array
+        indexes of image locations with NaN values
+    """
+    exptype = input_model.meta.exposure.type.lower()
+
+    # Initialize mask to all True. Subsequent operations will mask
+    # out pixels that contain signal.
+    # Note: Mask will be 3D for BOTS mode data
+    Mask = np.full(np.shape(input_model.dq), True)
+
+    # If IFU, mask all pixels contained in the IFU slices
+    if exptype == 'nrs_ifu':
+        Mask = mask_ifu_slices(input_model, Mask)
+
+    # If MOS or FS, mask all pixels affected by open slitlets
+    if exptype in ['nrs_fixedslit', 'nrs_brightobj', 'nrs_msaspec']:
+        Mask = mask_slits(input_model, Mask)
+
+    # If IFU or MOS, mask pixels affected by failed-open shutters
+    if exptype in ['nrs_ifu', 'nrs_msaspec']:
+        open_pix = np.where(input_model.dq & dqflags.pixel['MSA_FAILED_OPEN'])
+        Mask[open_pix] = False
+
+    # Temporarily reset NaN pixels and mask them.
+    # Save the list of NaN pixel coords, so that they can be reset at the end.
+    nan_pix = np.where(np.isnan(input_model.data))
+    input_model.data[nan_pix] = 0
+    Mask[nan_pix] = False
+
+    # If IFU or MOS, mask the fixed-slit area of the image; uses hardwired indexes
+    if exptype in ['nrs_ifu', 'nrs_msaspec']:
+        Mask[922:1116, :] = False
+
+    # Use left/right reference pixel columns (first and last 4). Can only be
+    # applied to data that uses all 2048 columns of the detector.
+    if Mask.shape[-1] == 2048:
+        Mask[..., :, :5] = True
+        Mask[..., :, -5:] = True  # keep one extra column on the right (always empty)
```

Comment:
The extra column on the right I added to the notebooks as a workaround for NSClean, since I couldn't modify the code.  If we add the catch for empty data in a column to the fit function noted above, the extra column shouldn't be needed here -- '-4:' should be sufficient.

Reply:
fixed in latest commit